packages/pnpm-sync-lib/src/utilities.ts

```diff
@@ -6,3 +10,40 @@
 export function pnpmSyncGetJsonVersion(): string {
   return require('../package.json').version;
 }
+
+export function getFilesInDirectory(directory: string, includeDirectory: boolean): Array<IFileStat> {
```

Comment:
How big is `Array<IFileStat>`?  For example, if we are copying 10,000 files, do we have to allocate memory to track 10,000 array elements? Or are `IFileStat` items created/disposed incrementally as the algorithm progresses?

Reply:
I feel like it is necessary to have array to hold all filenames. It will be easier to write the copy logic, and less corner cases. Also, from space perspective, we only save file metainfo to the array, so the total size should be relatively small.